src/modules/imselector/imselector.cpp

```diff
@@ -223,6 +244,21 @@ bool IMSelector::trigger(InputContext *inputContext, bool local) {
     return true;
 }
 
+bool IMSelector::selectInputMethod(InputContext *inputContext, size_t index,
+                                   bool local) {
+    auto &list =
+        instance_->inputMethodManager().currentGroup().inputMethodList();
+    if (index >= list.size())
+        return false;
+
+    // Use the same logic as selecting a input method from candidate lists. This
+    // will reset IMSelector when we select an input method via hotkey.
+    IMSelectorCandidateWord candidate_word(
+        this, instance_->inputMethodManager().entry(list[index].name()), local);
+    candidate_word.select(inputContext);
```

Comment:
This implementation looks weird. You should make candidate word use a common function, instead of creating a virtual candidate word.

Reply:
I choose this implementation because we already inject `IMSelector` into `IMSelectorCandidateWord`. Extracting that part of code to common utils doesn't fit well.

WDYT of removing `IMSelectorCandidateWord.IMSelector` and add a method `IMSelector::select(InputContext ic, IMSelectorCandidateWord word)`, which will call another new method `IMSelector::select(InputContext ic, string uniqueName, bool local)`. In this way we can replace this block with a shared method in `IMSelector` instead of having the selection logic in `IMSelectorCandidateWord`

Reply:
Just realized `select` is an interface of the base class of `IMSelectorCandidateWord`. Chose the common function approach you mentioned.